rskj-core/src/main/java/co/rsk/peg/Bridge.java

```diff
@@ -1605,8 +1605,8 @@ public static BridgeMethods.BridgeMethodExecutor activeAndRetiringFederationOnly
         return (self, args) -> {
             boolean isFromActiveFed = BridgeUtils.isFromFederateMember(self.rskTx, self.bridgeSupport.getActiveFederation(), self.signatureCache);
 
-            Federation retiringFederation = self.bridgeSupport.getRetiringFederation();
-            boolean isFromRetiringFed = retiringFederation != null && BridgeUtils.isFromFederateMember(self.rskTx, retiringFederation, self.signatureCache);
+            Optional<Federation> retiringFederation = self.bridgeSupport.getRetiringFederation();
+            boolean isFromRetiringFed = retiringFederation.isPresent() && BridgeUtils.isFromFederateMember(self.rskTx, retiringFederation.get(), self.signatureCache);
```

Comment:
Since these checks are mutually exclusive, what if we only check whether it is from a retiring fed when needed (when we know it is not from an active fed)? Wdyt?